cmake/External/nccl.cmake

```diff
@@ -39,34 +39,8 @@ if(NOT __NCCL_INCLUDED)
       INSTALL_COMMAND ""
       )
 
-    # Detect objcopy version
-    execute_process(COMMAND "${CMAKE_OBJCOPY}" "--version" OUTPUT_VARIABLE OBJCOPY_VERSION_STR)
-    string(REGEX REPLACE "GNU objcopy version ([0-9])\\.([0-9]+).*" "\\1" OBJCOPY_VERSION_MAJOR ${OBJCOPY_VERSION_STR})
-    string(REGEX REPLACE "GNU objcopy version ([0-9])\\.([0-9]+).*" "\\2" OBJCOPY_VERSION_MINOR ${OBJCOPY_VERSION_STR})
-
-    if((${OBJCOPY_VERSION_MAJOR} GREATER 2) OR ((${OBJCOPY_VERSION_MAJOR} EQUAL 2) AND (${OBJCOPY_VERSION_MINOR} GREATER 27)))
```

Comment:
This would increase binary size by 100+Mb (and context size as well)

So why is it needed? Device side linking should work for NCCL, isn't it? (In that case there should be a mention in release notes)

Reply:
Ditto - please check the comments above. Library slimming causes `undefined references` issues (also reported in https://github.com/NVIDIA/nccl/issues/639).

Reply:
Gentle ping - WDYT?

Reply:
I don't want binary size increase, therefore imo slimming should be fixed rather than removed.

Reply:
I see that you worked on this PR to strip out the `__nv_relfatbin` section a few years ago https://github.com/pytorch/pytorch/pull/35843 and wanted to keep it. Are you sure that this "slimming" optimization depends on documented stable behaviors that can work across NCCL versions?